### PR TITLE
String partitioner

### PR DESCRIPTION
**Pull request by gsvic**

### CODE REVIEW

File: src/main/java/io/tiledb/spark/StringPartitioner.java [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+  public StringPartitioner(int minChar, int maxChar) {
+    this.minChar = minChar;
+    this.maxChar = maxChar;
+  }
+
+  /**
+   * Computes the next possible string by incrementing the least possible character. The next string
+   * is limited to the input string length. For example, if the minimum character provided is 'z',
+   * then a string like `zzz` cannot be further incremented and it will be returned as it is. On the
+   * other hand, the next string of `aaz` will be `aba`, the next of `a` `b` and so on.
```
Comment (Shelnutt2) [inline src/main/java/io/tiledb/spark/StringPartitioner.java:28]: Add a note that your examples here are limited to a-z for simplicity of the example but go from 0-127 ascii characters in actual code.

File: src/main/java/io/tiledb/spark/StringPartitioner.java [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+   *
+   * @param a1 The left bound
+   * @param a2 The right bound
+   * @return The number of possible strings
+   */
+  public int distance(String a1, String a2) {
+    String tmp = a1;
+    int dist = 0;
+    while (tmp.compareTo(a2) < 0) {
+      // System.out.println(tmp);
```
Comment (Shelnutt2) [inline src/main/java/io/tiledb/spark/StringPartitioner.java:75]: remove commented out line

File: src/main/java/io/tiledb/spark/SubArrayRanges.java [hunk 3]
```diff
@@ -326,4 +334,11 @@ public boolean splittable() {
     double sum = widths.stream().map(e -> (Double) e).mapToDouble(Double::doubleValue).sum();
     return widths.stream().map(e -> ((Double) e) / sum).collect(Collectors.toList());
   }
+
+  public static void main(String[] args) {
```
Comment (Shelnutt2) [inline src/main/java/io/tiledb/spark/SubArrayRanges.java:338]: Should this main be removed?

**Comment by gsvic** [comment]
All comments resolved @Shelnutt2

**Pull request merged**